modules/basics/src/main/java/com/opengamma/strata/basics/currency/FxRate.java

```diff
@@ -236,6 +237,23 @@ public FxRate toConventional() {
 
   //-------------------------------------------------------------------------
   /**
+   * Returns an instance with the rate updated.
+   * <p>
+   * This returns an {@code FxRate} with the same currency pair and an updated rate.
+   * For example, the operator could be used to add the forward points of an FX swap.
+   * <pre>
+   *   forwardRate = nearRate.mapRate(rate -> rate + forwardPoints);
+   * </pre>
+   *
+   * @param operator  the operator to be applied to the existing rate
+   * @return a new instance with the rate updated
+   */
+  public FxRate mapRate(DoubleUnaryOperator operator) {
```

Comment:
I've removed the method. But really I feel we are making it too hard to access the rate.